src/Features/Core/Portable/CodeRefactorings/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -35,10 +35,12 @@ protected abstract class Analyzer<TBaseArgumentSyntax, TArgumentSyntax, TArgumen
                 }
 
                 var token = root.FindToken(context.Span.Start);
-                if (IsCloseParenOrComma(token))
+                if (token.Span.Start == context.Span.Start &&
+                    IsCloseParenOrComma(token))
                 {
-                    if (token.Span.Start != context.Span.Start ||
-                        IsCloseParenOrComma(token = token.GetPreviousToken()))
+                    token = token.GetPreviousToken();
+                    if (token.Span.End != context.Span.Start || 
+                        IsCloseParenOrComma(token))
```

Comment:
@CyrusNajmabadi It's a sneaky edge case. The missing argument led to the previous token being `,`, which is not part of the argument *list* but not the argument itself. This caused the code fix to search upwards and find the outer invocation and try to add the named argument there. It would be a rare but really weird experience when it occurred.